### PR TITLE
Support for classic era

### PR DESCRIPTION
**Pull request by RyadaProductions**
# Summary of Changes

Adds support (Finally) for Classic Era!!!!!!!!!!
1. Code wise had to start using a seperate API point for SetGradient.
2. Had to disable the BNet functions for the friends module.
3. Had to add classic only hearthstones.
4. Fixed multitude of errors due to a change in the API since the Classic ERA Api is ancient.

# Description

Finally now we can play Classic HC with a proper UI

# To test

-   [ ] Too much to list we basically either need to trial by fire or find a few people that actively play classic era enough to test this

### CODE REVIEW

File: Initialize.lua [hunk 1]
```diff
@@ -99,7 +99,7 @@ function TXUI:Initialize()
   end
 
   -- Check for non Wrath, non Retail
```
Comment (Toxicom) [inline Initialize.lua]: Update comment

File: Initialize.lua [hunk 2]
```diff
@@ -110,6 +110,11 @@ function TXUI:Initialize()
     I.InterruptSpellMap = I.InterruptSpellMap_Wrath
   end
 
+  if self.IsClassic then
+    I.HearthstoneData = I.HearthstoneData_Classic
+    I.InterruptSpellMap = I.InterruptSpellMap_Wrath
```
Comment (Toxicom) [inline Initialize.lua]: Are interrupts identical to Wrath? If so, we should leave a comment saying that

File: Modules/Installer/Core.lua [hunk 3]
```diff
@@ -164,7 +164,7 @@ function IS:ElvUIProfileDialog()
   local function createNewProfile(name)
     if strtrim(name) == "" then return end
 
-    if E.data:IsDualSpecEnabled() then
+    if not TXUI.IsClassic and E.data:IsDualSpecEnabled() then
```
Comment (Toxicom) [inline Modules/Installer/Core.lua:167]: Does classic not have dual spec? Was that a tbc/wrath feature?

**Review by Toxicom — APPROVED** [review]
As much as I understand it looks fine, just a couple tiny nitpick comments. Let's wait for @Stiimo cause they know more than me 😆

File: Core/Functions/Color.lua [hunk 4]
```diff
@@ -11,11 +11,21 @@ function F.Color.EqualToRGB(aColor, r, g, b)
 end
 
 function F.Color.SetGradient(obj, orientation, minColor, maxColor)
-  obj:SetGradient(orientation, minColor, maxColor)
+  if TXUI.IsClassic then
+    local r1, g1, b1, a1 = minColor:GetRGBA()
+    local r2, g2, b2, a2 = minColor:GetRGBA()
```
Comment (Stiimo) [inline Core/Functions/Color.lua]: Typo 
```suggestion
    local r2, g2, b2, a2 = maxColor:GetRGBA()
```

File: Initialize.lua (hunk 2 above)
Comment (RyadaProductions) [reply Initialize.lua]: Interrupt map is empty for Wrath specifically so figured reusing it was best. But I thought about changing the name to InterruptSpellMap_Empty

File: Modules/Installer/Core.lua (hunk 3 above)
Comment (Stiimo) [reply Modules/Installer/Core.lua:167]: It was wotlk feature, if I remember correctly

Comment (RyadaProductions) [reply Modules/Installer/Core.lua:167]: Dual spec was a Wrath feature. There was a lot of people complaining that they wanted it in classic and tbc but they didn't give it to us afaik.

File: Initialize.lua (hunk 2 above)
Comment (RyadaProductions) [reply Initialize.lua]: Updated the name to InterruptSpellMap_Empty, and on definition of the empty spellmap I added a 2 line comment to explain

**Review by Stiimo — APPROVED** [review]
Ran a fast test on Retail and Wrath, and had no bugs so far. Sadly, I don't have a Classic Era client, so can't test it there. Anyway, everything looks ok to me.

**Pull request merged**